src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,884 @@
+			self::get_dependency_api_data();
+		}
+
+		$dependencies         = self::get_dependencies( $plugin_file );
+		$dependency_filepaths = self::get_dependency_filepaths();
+
+		$dependency_names = array();
+		foreach ( $dependencies as $dependency ) {
+			if ( ! isset( $dependency_filepaths[ $dependency ] ) ) {
+				continue;
+			}
+
+			// Use the name if it's available, otherwise fall back to the slug.
+			if ( isset( self::$dependency_api_data[ $dependency ]['name'] ) ) {
+				$name = self::$dependency_api_data[ $dependency ]['name'];
+			} elseif ( isset( self::$plugin_dirnames[ $dependency ] ) ) {
+				$dependency_filepath = self::get_dependency_filepath( $dependency );
+				if ( false !== $dependency_filepath ) {
+					$name = self::$plugins[ $dependency_filepath ]['Name'];
+				}
+			} else {
+				$name = $dependency;
+			}
+
+			$dependency_names[ $dependency ] = $name;
+		}
+
+		return $dependency_names;
+	}
+
+	/**
+	 * Gets the filepath for a dependency, relative to the plugin's directory.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $slug The dependency's slug.
+	 * @return string|false If installed, the dependency's filepath relative to the plugins directory, otherwise false.
+	 */
+	public static function get_dependency_filepath( $slug ) {
+		if ( empty( self::$dependency_filepaths ) ) {
+			self::get_dependency_filepaths();
+		}
+
+		return self::$dependency_filepaths[ $slug ];
+	}
+
+	/**
+	 * Returns API data for the dependency.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $slug The dependency's slug.
+	 * @return array|false The dependency's API data on success, otherwise false.
+	 */
+	public static function get_dependency_data( $slug ) {
+		if ( empty( self::$dependency_api_data ) ) {
+			self::get_dependency_api_data();
+		}
+
+		if ( isset( self::$dependency_api_data[ $slug ] ) ) {
+			return self::$dependency_api_data[ $slug ];
+		}
+
+		return false;
+	}
+
+	/**
+	 * Displays an admin notice if dependencies are not installed.
+	 *
+	 * @since 6.4.0
+	 */
+	public static function display_admin_notice_for_unmet_dependencies() {
+		/*
+		 * Plugin deactivated if dependencies not met.
+		 * Transient on a 10 second timeout.
+		 */
+		$deactivate_requires = get_site_transient( 'wp_plugin_dependencies_deactivated_plugins' );
+		if ( ! empty( $deactivate_requires ) ) {
+			$deactivated_plugins = '';
+			foreach ( $deactivate_requires as $deactivated ) {
+				$deactivated_plugins .= '<li>' . esc_html( self::$plugins[ $deactivated ]['Name'] ) . '</li>';
+			}
+			wp_admin_notice(
+				sprintf(
+					/* translators: 1: plugin names */
+					__( 'The following plugin(s) have been deactivated due to uninstalled or inactive dependencies: %s' ),
+					"<ul>$deactivated_plugins</ul>"
+				),
+				array(
+					'type'        => 'error',
+					'dismissible' => true,
+				)
+			);
+		} else {
+			// More dependencies to install.
+			$installed_slugs = array_map( 'dirname', array_keys( self::$plugins ) );
```

Comment:
Need to handle single file plugins here.
```suggestion
			$installed_slugs = array();
			foreach ( array_keys( self::$plugins ) as $plugin ) {
				$installed_slugs[] = str_contains( $plugin, '/' ) ? dirname( $plugin ) : $plugin;
			}
```